nobo.py

```diff
@@ -330,54 +355,67 @@ def create_override(self, mode, type, target_type, target_id='-1', end_time='-1'
-                    dicti = dict(zip(self.API.STRUCT_KEYS_COMPONENT, r[1:]))
-                    self.components[dicti['serial']] = dicti
-                    logging.info('added/updated component: %s', dicti['name'])
-
-                elif r[0] in [self.API.RESPONSE_WEEK_PROFILE_INFO, self.API.RESPONSE_UPDATE_V02]:
-                    dicti = dict(zip(self.API.STRUCT_KEYS_WEEK_PROFILE, r[1:]))
-                    dicti['profile'] = r[-1].split(',')
-                    self.week_profiles[dicti['week_profile_id']] = dicti
-                    logging.info('added/updated week profile: %s', dicti['name'])
-
-                elif r[0] in [self.API.RESPONSE_OVERRIDE_INFO, self.API.RESPONSE_ADD_B03]:
-                    dicti = dict(zip(self.API.STRUCT_KEYS_OVERRIDE, r[1:]))
-                    self.overrides[dicti['override_id']] = dicti
-                    logging.info('added/updated override: id %s', dicti['override_id'])
-
-                elif r[0] in [self.API.RESPONSE_STATIC_INFO, self.API.RESPONSE_UPDATE_V03]:
-                    self.hub_info = dict(zip(self.API.STRUCT_KEYS_HUB, r[1:]))
-                    logging.info('updated hub info: %s', self.hub_info)
-                    if r[0] == self.API.RESPONSE_STATIC_INFO:
-                        self.all_received = True
-
-                #TODO: Add all the other response_remove commands
-                elif r[0] == self.API.RESPONSE_REMOVE_S03:
-                    dicti = dict(zip(self.API.STRUCT_KEYS_OVERRIDE, r[1:]))
-                    popped_override = self.overrides.pop(dicti['override_id'], None)
-                    logging.info('removed override: id%s', dicti['override_id'])
-
-                elif r[0][0] == 'E':
-                    logging.error('error! what did you do? %s', r)
-                    #TODO: Raise something here?
+        while not self.socket_receive_exit_flag.is_set():
+            if self.socket_connected.is_set():
+                resp = self.get_response(True)
+                for r in resp:
+                    logging.debug('received: %s', r)
 
-                else:
-                    logging.warning('behavior undefined for this command: {}'.format(r))
-                    warnings.warn('behavior undefined for this command: {}'.format(r)) #overkill?        
+                    if r[0] == [self.API.HANDSHAKE]:
+                        pass # Handshake, no action needed
+
+                    elif r[0][0] == 'E':
+                        logging.error('error! what did you do? %s', r)
+                        #TODO: Raise something here?
+
+                    else:
+                        self.response_handler(r)
+
+            else:
+                self.reconnect_hub()
+
+        logging.info('receive thread exited')
+
+    def response_handler(self, r):
+        # A lot of info is incoming, will end with RESPONSE_STATIC_INFO
+        if r[0] == self.API.RESPONSE_SENDING_ALL_INFO:
+            self.socket_received_all_info.clear()
```

Comment:
Should we clear existing dictionaries at this point? Some items may have been removed.